### PR TITLE
Handle multiple file system: s3/local

### PR DESCRIPTION
**Pull request by rounak610**
### Description

Handle multiple file system: s3/local

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: app/services/filestore/impl/local.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		return
+	}
+	content = io.NopCloser(bytes.NewReader(fileContent))
+	return content, nil
+}
+
+func (lfs LocalFileStore) ReadFileWithInfo(path string) (content io.ReadCloser, contentLength int64, contentType string, err error) {
+    filePath, err := lfs.getFilePath(path)
+    if err != nil {
+        return nil, 0, "", fmt.Errorf("failed to get file path: %w", err)
```
Comment (czezi) [inline app/services/filestore/impl/local.go:68]: Use pointers to nullable return type, make it 
```
contentType *string
```

File: app/services/story_service.go [hunk 2]
```diff
@@ -689,27 +699,49 @@ func (s *StoryService) UpdateStoryStatusWithTx(tx *gorm.DB, storyId int, progres
 	return s.storyRepo.UpdateStoryStatusWithTx(tx, storyId, progress)
 }
 
+func (s *StoryService) GetImageReaderByStoryId(storyId int) (io.ReadCloser, int64, string, error) {
+    filePath, err := s.storyFileRepo.GetFilePathByStoryID(storyId)
+    if err != nil {
+        return nil, 0, "", fmt.Errorf("error occurred while getting file path: %w", err)
+    }
+
+	reader, contentLength, contentType, err := s.fileStore.ReadFileWithInfo(filePath)
```
Comment (czezi) [inline app/services/story_service.go:708]: Fix indentation

File: app/services/story_service.go [hunk 3]
```diff
@@ -689,27 +699,49 @@ func (s *StoryService) UpdateStoryStatusWithTx(tx *gorm.DB, storyId int, progres
...
+    if err != nil {
+        return nil, 0, "", fmt.Errorf("error reading file: %w", err)
+    }
+
+	ext := strings.ToLower(filepath.Ext(filePath))
```
Comment (czezi) [inline app/services/story_service.go:713]: fix indentation

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 4]
```diff
@@ -49,9 +49,9 @@ func NewOpenAINextJsCodeGenerationExecutor(
 		storyService:         storyService,
 		activityLogService:   activityLogService,
 		designReviewService:  designReviewService,
-		s3Service:            s3Service,
 		llmAPIKeyService:     llmAPIKeyService,
 		logger:               logger,
+		fileStore: 			  fileStore,
```
Comment (czezi) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:54]: Fix indentation

File: workspace-service/app/services/impl/docker_job_service.go [hunk 5]
```diff
@@ -56,6 +57,11 @@ func (js DockerJobService) CreateJob(request dto.CreateJobRequest) (res *dto.Cre
 					Source: js.config.VolumeSource(),
 					Target: js.config.VolumeTarget(),
 				},
+				{
+					Type:   mount.TypeVolume,
+					Source: "supercoder_filestore",
```
Comment (czezi) [inline workspace-service/app/services/impl/docker_job_service.go:62]: move this to config

**Review by czezi — CHANGES_REQUESTED** [review]

File: workspace-service/app/services/impl/docker_job_service.go (hunk 5 above)
Comment (rounak610) [reply workspace-service/app/services/impl/docker_job_service.go]: fixed

File: app/services/story_service.go [hunk 6]
```diff
@@ -33,11 +35,11 @@ type StoryService struct {
 	storyInstructionRepo   *repositories.StoryInstructionRepository
 	asynqClient            *asynq.Client
 	logger                 *zap.Logger
-	s3Service              *s3_providers.S3Service
 	storyFileRepo          *repositories.StoryFileRepository
 	hashIdGenerator        *utils.HashIDGenerator
 	workspaceServiceClient *workspace.WorkspaceServiceClient
 	projectService         *ProjectService
+	fileStore 			   filestore.FileStore
```
Comment (luciferlinx101) [inline app/services/story_service.go:42]: correct identation

File: app/services/story_service.go [hunk 7]
```diff
@@ -689,27 +699,51 @@ func (s *StoryService) UpdateStoryStatusWithTx(tx *gorm.DB, storyId int, progres
...
 		storyTestCaseRepo:      storyTestCaseRepo,
 		storyInstructionRepo:   storyInstructionRepo,
 		asynqClient:            asynqClient,
 		logger:                 logger.Named("StoryService"),
-		s3Service:              s3Service,
 		storyFileRepo:          storyFileRepo,
 		hashIdGenerator:        utils.NewHashIDGenerator(5),
 		workspaceServiceClient: workspaceServiceClient,
 		projectService:         projectService,
+		fileStore: 				fileStore,
```
Comment (luciferlinx101) [inline app/services/story_service.go:747]: correct indentation

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 8]
```diff
@@ -26,9 +26,9 @@ type OpenAiNextJsCodeGenerator struct {
 	storyService         *services.StoryService
 	activityLogService   *services.ActivityLogService
 	designReviewService  *services.DesignStoryReviewService
-	s3Service            *s3_providers.S3Service
 	llmAPIKeyService     *services.LLMAPIKeyService
 	logger               *zap.Logger
+	fileStore 	      filestore.FileStore
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:31]: why is this not pointer? like * filestore.FileStore ?

File: server.go [hunk 9]
```diff
@@ -485,6 +518,7 @@ func main() {
 
 		story.GET("/code", storiesController.GetCodeForDesignStory)
 		story.GET("/design", storiesController.GetDesignStoryByID)
+		story.GET("/fetch_image", storiesController.GetImageByStoryId)
```
Comment (luciferlinx101) [inline server.go:521]: can you use kebab case instead of snake case as in other endpoints

File: gui/src/app/(programmer)/design/review/[story_id]/page.tsx [hunk 10]
```diff
@@ -187,11 +187,11 @@ const ReviewPage: React.FC = (props) => {
                   }
                 >
                   <Image
-                    src={story.input_file_url}
+                    src={`/api/stories/${story_id}/fetch_image`}
```
Comment (luciferlinx101) [inline gui/src/app/(programmer)/design/review/[story_id]/page.tsx:190]: this will need change in case endpoint changes are made

File: gui/src/components/DesignStoryComponents/DesignStoryDetails.tsx [hunk 11]
```diff
@@ -219,9 +219,9 @@ const DesignStoryDetails: React.FC<DesignStoryDetailsProps> = ({
             <div
               className={`relative flex max-h-[50vh] justify-center overflow-hidden rounded-lg ${styles.story_image_container}`}
             >
-              <img
-                className={'max-h-full max-w-full object-contain'}
-                src={selectedStory.input_file_url}
+              <CustomImage
+                className={'size-max'}
+                src={`/api/stories/${selectedStory.id}/fetch_image`}
```
Comment (luciferlinx101) [inline gui/src/components/DesignStoryComponents/DesignStoryDetails.tsx:224]: here also need to update

File: gui/src/components/DesignStoryComponents/DesignStoryList.tsx [hunk 12]
```diff
@@ -59,26 +61,26 @@ const DesignStoryList: React.FC = () => {
               >
                 <div className={'relative h-full w-full'}>
                   <Image
-                      src={story.input_file_url}
-                      alt={'design_image'}
-                      fill
-                      className="object-contain"
-                      loading="lazy"
+                    src={`/api/stories/${story.id}/fetch_image`}
```
Comment (luciferlinx101) [inline gui/src/components/DesignStoryComponents/DesignStoryList.tsx:64]: Here also need to update

File: gui/src/components/DesignStoryComponents/ReviewList.tsx [hunk 13]
```diff
@@ -49,11 +49,11 @@ const ReviewList: React.FC = () => {
                     className={`${styles.review_image_container} relative flex h-[60px] w-[60px] items-center justify-center overflow-hidden rounded-lg`}
                   >
                     <Image
-                      src={story.input_file_url}
+                      src={`/api/stories/${story.id}/fetch_image`}
                       alt={'design_image'}
                       fill
-                      className="object-contain"
-                      loading="lazy"
```
Comment (luciferlinx101) [inline gui/src/components/DesignStoryComponents/ReviewList.tsx:56]: why are these changing lets stick to one standard

File: gui/src/components/WorkBenchComponents/DesignWorkbench.tsx [hunk 14]
```diff
@@ -83,9 +84,9 @@ const ActiveDesignWorkbench: React.FC<DesignWorkbenchProps> = ({
             style={{ height: 'calc(100% - 34px)', overflow: 'hidden' }}
           >
             {selectedStory && (
-              <img
-                src={selectedStory.input_file_url}
-                alt="input_image"
+              <CustomImage
+                src={`/api/stories/${selectedStoryId}/fetch_image`}
```
Comment (luciferlinx101) [inline gui/src/components/WorkBenchComponents/DesignWorkbench.tsx:88]: Here also it will need update

File: gui/src/components/WorkBenchComponents/DesignWorkbench.tsx [hunk 15]
```diff
@@ -83,9 +84,9 @@ const ActiveDesignWorkbench: React.FC<DesignWorkbenchProps> = ({
...
+                alt={'input_image'}
```
Comment (luciferlinx101) [inline gui/src/components/WorkBenchComponents/DesignWorkbench.tsx:89]: why is this changed?

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

File: app/config/filestore.go [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+	return fsc.config.String("filestore.type")
+}
+
+func (fsc *FileStoreConfig) GetLocalDir() string {
+	return fsc.config.String("filestore.local.dir")
+}
+
+func (fsc *FileStoreConfig) GetS3Bucket() string {
+	return fsc.config.String("filestore.s3.bucket")
+}
```
[the remark this answers is not in the document]
Comment (rounak610) [reply app/config/filestore.go]: fixed

File: README.md [hunk 17]
```diff
@@ -115,13 +115,10 @@ If you are using a shell other than bash, replace bash with your specific shell
 The basic environment variables are included in the Docker setup, making initial configuration quick and easy. However, for using Amazon S3 to store images for the Design Section, you need to configure the following environment variables manually:
 ```bash
 
-export AI_DEVELOPER_AWS_ACCESS_KEY_ID=
-export AI_DEVELOPER_AWS_SECRET_ACCESS_KEY=
-export AI_DEVELOPER_AWS_BUCKET_NAME=
-export AI_DEVELOPER_AWS_REGION=
+export AI_DEVELOPER_FILESTORE_TYPE="local"
```
[the remark this answers is not in the document]
Comment (rounak610) [reply README.md]: fixed

File: app/services/story_service.go [hunk 18]
```diff
@@ -689,27 +699,51 @@ func (s *StoryService) UpdateStoryStatusWithTx(tx *gorm.DB, storyId int, progres
...
 	storyRepo *repositories.StoryRepository,
 	storyTestCaseRepo *repositories.StoryTestCaseRepository,
 	storyInstructionRepo *repositories.StoryInstructionRepository,
 	asynqClient *asynq.Client,
 	logger *zap.Logger,
-	s3Service *s3_providers.S3Service,
 	storyFileRepo *repositories.StoryFileRepository,
 	workspaceServiceClient *workspace.WorkspaceServiceClient,
 	projectService *ProjectService,
+	fileStore filestore.FileStore,
```
Comment (luciferlinx101) [inline app/services/story_service.go:735]: why is it not *filestore.FileStore?

File: executor.go [hunk 19]
```diff
@@ -58,9 +60,56 @@ func main() {
...
+
+	if err = c.Provide(func(
+		awsConfig *config.AWSConfig,
+		storeConfig *config.FileStoreConfig,
+		localFileStoreConfig *config.LocalFileStoreConfig,
+		s3fileStoreConfig *config.S3FileStoreConfig,
+		awsSession *session.Session,
+		logger *zap.Logger,
+	) filestore.FileStore {
+		if storeConfig.GetFileStoreType() == "local" {
```
Comment (luciferlinx101) [inline executor.go:102]: local we can move it to consts

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

**Review by luciferlinx101 — APPROVED** [review]
